beacon-chain/cache/signed_execution_header.go

```diff
@@ -0,0 +1,87 @@
+package cache
+
+import (
+	"bytes"
+	"sync"
+
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	enginev1 "github.com/prysmaticlabs/prysm/v5/proto/engine/v1"
+)
+
+var (
+	// This cache is intended for use by the sync service to store signed execution payload headers after they pass validation.
+	// The signed header from this cache could be used by the proposer when proposing the next slot.
+	cachedSignedExecutionPayloadHeader = make(map[primitives.Slot][]*enginev1.SignedExecutionPayloadHeader)
```

Comment:
Why another global cache instead of a structure one held by the service? Take a look at the PayloadAttestationCache  example